Pixel painting sample Hydra DApp

abailly-iohk opened the pull request:
document the outcome of Team's Hackathon in Lyon as a sample basic web app using a Hydra Head to allow collaborative pixel painting

KtorZ committed:
Add pixel-painting bootstrap app.

ch1bo committed:
Add a canvas and drawPixel

Also some way to launch a local http server

KtorZ committed:
Decode transaction's auxiliary data and paint call 'drawPixel'

ch1bo committed:
Add a style sheet

ch1bo committed:
Automatically scale canvas

ch1bo committed:
Listen for canvas events and drawPixels directly

abailly committed:
Add small exe to submit a Pixel as metadata

KtorZ committed:
Add simple pixel-painting webserver, needs to wire into Arnaud's

ch1bo committed:
Add a color picker and use it when drawing pixels

KtorZ committed:
Integrate pixel painter into HTTP server.

KtorZ committed:
Integrate canvas into pixel-painting server.

KtorZ committed:
Fix ENV variable for pixel-painting server

ch1bo committed:
Not gitignore index.html

ch1bo committed:
Add some instructions to pixel-painting

KtorZ committed:
Handle CORS on the little webserver.

abailly committed:
Remove copy of Painter code

KtorZ committed:
Empty message queue before querying.

abailly committed:
Set metadata type to 14

ch1bo committed:
Listen on 0.0.0.0

KtorZ committed:
Fix UTXO parsing and ENV var (network host) parsing.

ch1bo committed:
Hard-code networkId (as only 'Testnet' bit is used)

KtorZ committed:
Fix CBOR decoder for auxiliary data.

abailly committed:
Serve static assets from pixel-painting app

This requires `cabal run pixel-painting` to be run inside the
`pixel-painting` directory

abailly committed:
Update pixel-painting instructions

ch1bo committed:
Use same host for fetch and update instructions

ch1bo committed:
Drop the now simple-http-server

KtorZ committed:
Add some more style to make it a real DApp.

ch1bo committed:
Fix js click event offset

KtorZ committed:
bridge websocket through the local webserver -> remove need to hardcode host on client.

KtorZ committed:
Add optional query parameter '?delay=<ms>' for deferring pixel painting.

KtorZ committed:
Fix canvas size on mobile.

KtorZ committed:
Create ws connection once, and flush only end of queue when painting.

KtorZ committed:
Rename to 'Hydraw'

KtorZ committed:
Add favicon.

abailly committed:
Basic documentation of pixel-painter app #269

KtorZ commented on pixel-painting/bundle.js:
```diff
@@ -0,0 +1,83 @@
+  .split('&')
+  .filter(x => x != '')
+  .map(x => x.split('='))
+const delay = Number((query.filter(([k, v]) => k == 'delay')[0] || [])[1] || 0);
```
How does one remove himself from commit authors? Asking for a friend.

KtorZ commented on pixel-painting/pixel-painting.cabal:
```diff
@@ -0,0 +1,105 @@
+cabal-version: 2.2
+name:          pixel-painting
```
small note, but we should probably rename the app the whole folder as `Hydraw`

abailly committed:
Add disclaimer block #269

This tries to use a hack to display colored text in the README, a
feature currently unsupported by GitHub.

abailly committed:
Rename to hydraw #269

KtorZ commented on pixel-painting/README.md:
```diff
@@ -6,6 +6,13 @@ _Hydraw_ is the result of a 3 days Hackathon that took place in Lyon March 2022.
+- DISCLAIMER
+! This code is the result of an intense 3 days long Hackathon and is not meant for long-term maintenance and
+! evolution, hence it should not be taken as representative of the coding practices the Hydra team members
+! endorse in the rest of the codebase.
```
#YOLO

KtorZ commented on hydraw/README.md:
````diff
@@ -0,0 +1,40 @@
+
+``` javascript
+const HYDRA_API_HOST = "hydra.example.io:4001";
+```
````
This is not longer accurate, with the websocket bridge there's no more editing in bundle.js to do.

KtorZ committed:
Remove no longer accurate instructions from Hydraw's README.
